### PR TITLE
General improvements

### PR DESCRIPTION
**Pull request by RGarrido03**
Este PR é como a Worten, tem tudo e mais não sei o quê 💁‍♂️

No, seriously, this PR makes several general improvements.

### Common
- Add shadcn/ui elements to project, such as Form, Input and Label
- Make variant `background` of shadcn/ui's button become darker on hover
- Optimize and refactor code

### Insights page
- Make it responsive
- Reduce the size (i.e., the scale)
- Refactor elements to shadcn/ui components
- Use arrays with information instead of repeating elements

### Login page
- Add form validation
- Reduce the size (i.e., the scale)
- Refactor elements to shadcn/ui components
- Refactor SVGs as components with `currentColor` fill
- Refactor the title cover by using the original image
- Use arrays with information instead of repeating elements
- Use Next.js's `<Image />` component

## Data generator
- Change generation rate to 5 seconds in development and 60 seconds in production

## Backend
- Remove some unused imports

## VS Code settings
- Add default formatter for Java and XML files
- Add type checking for Python files

### CODE REVIEW

File: frontend/src/app/insight/page.tsx
Comment (falcaodiogo) [inline frontend/src/app/insight/page.tsx:1]: Whoa, the codee is soooo much cleaner now. Thank u!

File: frontend/src/app/login/form.tsx
Comment (falcaodiogo) [inline frontend/src/app/login/form.tsx:1]: Thank u @RGarrido03 and @zod

File: frontend/src/app/login/page.tsx
Comment (falcaodiogo) [inline frontend/src/app/login/page.tsx:1]: yup, i should have done that by now, much better this way!

**Review by falcaodiogo — APPROVED** [review]
Very good work. You not only made good critical decisions related to the production and development code settings, but also changed the code to be amazingly tight and packed. The use of Zod was also a very good ideia (form validation, yeaaa!!)

File: frontend/src/app/login/form.tsx
Comment (zod) [reply frontend/src/app/login/form.tsx:1]: Glad I could help ;)

**Pull request merged**